apps/judicial-system/backend/src/app/modules/file/file.service.ts

```diff
@@ -359,8 +398,9 @@ export class FileService {
   ): Promise<string> {
     if (
       isIndictmentCase(theCase.type) &&
-      hasIndictmentCaseBeenSubmittedToCourt(theCase.state) &&
-      file.category === CaseFileCategory.INDICTMENT
+      ((file.category === CaseFileCategory.INDICTMENT &&
+        hasIndictmentCaseBeenSubmittedToCourt(theCase.state)) ||
+        file.category === CaseFileCategory.RULING)
```

Comment:
When confirming a ruling the case must be completed. So, just like when getitng the file as an object from S3 you need to check that the case is completed when getting a signed url from S3.